File: trunk/incubator/net.sf.j2s.java.org.eclipse.osgi/src/org/eclipse/core/runtime/adaptor/StreamManagerOutputStream.java
/*******************************************************************************
 * Copyright (c) 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.core.runtime.adaptor;

import java.io.*;

/**
 * Represents an <code>OutputStream</code> provided by FileManager
 * @see StreamManager#getOutputStream(String)
 * <p>
 * Clients may not extend this class.
 * </p>
 * @since 3.1
 */
public class StreamManagerOutputStream extends FilterOutputStream {
	private String target;
	private StreamManager manager;
	private File outputFile;
	private int state;
	private StreamManagerOutputStream[] streamSet = null;

	StreamManagerOutputStream(OutputStream out, StreamManager manager, String target, File outputFile, int state) {
		super(out);
		this.manager = manager;
		this.target = target;
		this.outputFile = outputFile;
		this.state = state;
	}

	/** 
	 * Instructs this output stream to be closed and file manager to 
	 * be updated as appropriate.
	 * 
	 * @see StreamManager#getOutputStream(String)
	 * @see StreamManager#getOutputStreamSet(String[])
	 */
	public void close() throws IOException {
		manager.closeOutputStream(this);
	}

	/**
	 * Instructs this output stream to be closed and the contents removed.
	 * @see StreamManager#getOutputStream(String)
	 * @see StreamManager#getOutputStreamSet(String[])
	 */
	public void abort() {
		manager.abortOutputStream(this);
	}

	OutputStream getOutputStream() {
		return out;
	}

	String getTarget() {
		return target;
	}

	File getOutputFile() {
		return outputFile;
	}

	int getState() {
		return state;
	}

	void setState(int state) {
		this.state = state;
	}

	void setStreamSet(StreamManagerOutputStream[] set) {
		streamSet = set;
	}

	StreamManagerOutputStream[] getStreamSet() {
		return streamSet;
	}
}

File: old/incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/core/runtime/adaptor/StreamManagerOutputStream.js
Clazz.declarePackage ("org.eclipse.core.runtime.adaptor");
Clazz.load (["java.io.FilterOutputStream"], "org.eclipse.core.runtime.adaptor.StreamManagerOutputStream", null, function () {
c$ = Clazz.decorateAsClass (function () {
this.target = null;
this.manager = null;
this.outputFile = null;
this.state = 0;
this.streamSet = null;
Clazz.instantialize (this, arguments);
}, org.eclipse.core.runtime.adaptor, "StreamManagerOutputStream", java.io.FilterOutputStream);
Clazz.makeConstructor (c$, 
function (out, manager, target, outputFile, state) {
Clazz.superConstructor (this, org.eclipse.core.runtime.adaptor.StreamManagerOutputStream, [out]);
this.manager = manager;
this.target = target;
this.outputFile = outputFile;
this.state = state;
}, "java.io.OutputStream,org.eclipse.core.runtime.adaptor.StreamManager,~S,java.io.File,~N");
Clazz.overrideMethod (c$, "close", 
function () {
this.manager.closeOutputStream (this);
});
Clazz.defineMethod (c$, "abort", 
function () {
this.manager.abortOutputStream (this);
});
Clazz.defineMethod (c$, "getOutputStream", 
function () {
return this.out;
});
Clazz.defineMethod (c$, "getTarget", 
function () {
return this.target;
});
Clazz.defineMethod (c$, "getOutputFile", 
function () {
return this.outputFile;
});
Clazz.defineMethod (c$, "getState", 
function () {
return this.state;
});
Clazz.defineMethod (c$, "setState", 
function (state) {
this.state = state;
}, "~N");
Clazz.defineMethod (c$, "setStreamSet", 
function (set) {
this.streamSet = set;
}, "~A");
Clazz.defineMethod (c$, "getStreamSet", 
function () {
return this.streamSet;
});
});
